ng2-components/ng2-activiti-processlist/src/components/activiti-start-process.component.spec.ts

```diff
@@ -0,0 +1,143 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentFixture, TestBed, async } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { DebugElement } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+import { ActivitiFormModule } from 'ng2-activiti-form';
+import { AlfrescoTranslationService, CoreModule } from 'ng2-alfresco-core';
+import { FormService } from 'ng2-activiti-form';
+import { TranslationMock } from './../assets/translation.service.mock';
+import { ActivitiStartProcessButton } from './activiti-start-process.component';
+import { ActivitiProcessService } from '../services/activiti-process.service';
+
+describe('ActivitiStartProcessButton', () => {
+
+    let componentHandler: any;
+    let component: ActivitiStartProcessButton;
+    let fixture: ComponentFixture<ActivitiStartProcessButton>;
+    let processService: ActivitiProcessService;
+    let formService: FormService;
+    let getDefinitionsSpy: jasmine.Spy;
+    let startProcessSpy: jasmine.Spy;
+    let de: DebugElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [ CoreModule, ActivitiFormModule ],
+            declarations: [
+                ActivitiStartProcessButton
+            ],
+            providers: [
+                { provide: AlfrescoTranslationService, useClass: TranslationMock },
+                ActivitiProcessService,
+                FormService
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+
+        fixture = TestBed.createComponent(ActivitiStartProcessButton);
+        component = fixture.componentInstance;
+        processService = fixture.debugElement.injector.get(ActivitiProcessService);
+        formService = fixture.debugElement.injector.get(FormService);
+
+        getDefinitionsSpy = spyOn(processService, 'getProcessDefinitions').and.returnValue(Observable.of([{
+            id: 'my:process1',
+            name: 'My Process 1'
+        }, {
+            id: 'my:process2',
+            name: 'My Process 2'
+        }]));
+        startProcessSpy = spyOn(processService, 'startProcess').and.returnValue(Observable.of({
+            id: '32323',
+            name: 'Process'
+        }));
+
+        componentHandler = jasmine.createSpyObj('componentHandler', [
+            'upgradeAllRegistered',
+            'upgradeElement'
+        ]);
+        window['componentHandler'] = componentHandler;
+    });
+
+    it('should display the correct number of processes in the select list', () => {
+        fixture.detectChanges();
+        de = fixture.debugElement.query(By.css('select'));
```

Comment:
you can add   debug = fixture.debugElement;  in the beforeEach instead to assign it over every test case